lib/exception_handling.rb

```diff
@@ -192,7 +192,7 @@ def log_error(exception_or_string, exception_context = '', controller = nil, tre
     # Write an exception out to the log file using our own custom format.
     #
     def write_exception_to_log(ex, exception_context, timestamp, log_context = {})
-      ActiveSupport::Deprecation.silence do
+      with_deprecations_silenced do
```

Comment:
We'd discussed during planning that we might not actually need to silence deprecation warnings here. Do you know what we're silencing?

Reply:
It's protecting against potential feedback loops where we're logging a deprecation error, which then causes another deprecation error which we want to catch and log. So it's specifically for when the deprecation mode is set to `:error` instead of `:warn`